### PR TITLE
feat(ProfileShowase): Save action with the new backend and integration with backend

### PR DESCRIPTION
**Pull request by noeliaSD**
Closes [#13799](https://github.com/status-im/status-desktop/issues/13799)

### What does the PR do

- Created JSON file according to new backend structure when saving.
- Updated `dirty state` to sync writable and movable models when position is changed to have a better internal models sync.
- Reenabled identity fields save.

**NOTE: Only UI. It needs backend integration task #13688**

### Affected areas

Profile showcase

### Screenshot of functionality

https://github.com/status-im/status-desktop/assets/97019400/93967f0e-f0d5-4523-89f1-ee177bf754dc

### CODE REVIEW

File: ui/app/AppLayouts/Profile/helpers/ProfileShowcaseDirtyState.qml [hunk 1]
```diff
@@ -65,6 +61,18 @@ QObject {
 
     function changePosition(from, to) {
         visible.move(from, to)
+
+        // Sync writable with movable new positions:
+        const newOrder = visible.order()
+        let writableIndexes = []
+
+        for (var i = 0; i < newOrder.length; i++) {
+            writableIndexes.push(visibleSFPM.mapFromSource(newOrder[i]))
```
[the remark this answers is not in the document]
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/helpers/ProfileShowcaseDirtyState.qml]: Updated here

**Comment by noeliaSD** [comment]
I've started the integration with backend task in here. 

First issue found (in the moment `save` is clicked): 
```
Fatal: Object 0x60000692e3c0 destroyed while one of its QML signal handlers is in progress.
Most likely the object was deleted synchronously (use QObject::deleteLater() instead), or the application is running a nested event loop.
This behavior is NOT supported!
qrc:/app/AppLayouts/Profile/views/MyProfileView.qml:78: function() { [native code] } (:0, )
SIGABRT: Abnormal termination.
make: *** [run-macos] Abort trap: 6
```

**Comment by noeliaSD** [comment]
> I've started the integration with backend task in here.
> 
> First issue found (in the moment `save` is clicked):
> 
> ```
> Fatal: Object 0x60000692e3c0 destroyed while one of its QML signal handlers is in progress.
> Most likely the object was deleted synchronously (use QObject::deleteLater() instead), or the application is running a nested event loop.
> This behavior is NOT supported!
> qrc:/app/AppLayouts/Profile/views/MyProfileView.qml:78: function() { [native code] } (:0, )
> SIGABRT: Abnormal termination.
> make: *** [run-macos] Abort trap: 6
> ```

@MishkaRogachev solved the crash in his pr (base pr of this one). Here the final testing with the integration:

https://github.com/status-im/status-desktop/assets/97019400/bae4e0be-3b9c-47b5-b38e-2618744b37b6

**Review by MishkaRogachev — APPROVED** [review]
Great work, thanks! 
Tested, works fine

**Review by alexandraB99 — APPROVED** [review]

**Review by alexandraB99 — COMMENTED** [review]
please resolve conflicts

**Review by alexjba — APPROVED** [review]
LGTM!

**Comment by noeliaSD** [comment]
I'll push this branch to the base one an then @MishkaRogachev will integrate on master!

**Comment by alexjba** [comment]
We'll have to review the `reset` to fix that intermediary state when saving. Now the `save` is async and it's expected for the models to properly restore the dirty state whenever the source model is updated with identical data and remove the necessity to `reset` after `save`.

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml [hunk 2]
```diff
@@ -126,6 +122,11 @@ SettingsContentBase {
         id: priv
 
         property bool hasAnyProfileShowcaseChanges: showcaseModels.dirty
+        property bool isIdentityTabDirty: (!descriptionPanel.isEnsName &&
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/MyProfileView.qml:125]: `readonly` (the above too imho)

**Review by caybro — APPROVED** [review]
LGTM

Just one minor thing

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml (hunk 2 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml:125]: I'll push a fix next week for this!

**Pull request merged**